Mute Banner for Profile & Prevent Self-Target & Correctly Expire Early Unmutes

Ayymoss opened the pull request:
Add a temp/mute banner to their profile. (Priority lowered so as not to overwrite temp/ban banners)
Let the user know they cannot target themselves. 
Displays mute expiration on penalty if unmuted early.
Added condition to prevent expiration unmute spam.

```
COMMANDS_DENY_SELF_TARGET = You cannot target yourself

WEBFRONT_PROFILE_MUTE_INFO = Temporarily muted for {{reason}} with {{time}} remaining
WEBFRONT_PROFILE_TEMPMUTE_INFO = Permanently muted for {{reason}}
```

Ayymoss committed:
Fix self-targeting
Remove creation of penalty on mute expiration

Ayymoss committed:
Fix self-targeting
Remove creation of penalty on mute expiration

Ayymoss committed:
Display mute penalties on profile
Expire mute penalties on unmute

Ayymoss committed:
Display mute penalties on profile
Expire mute penalties on unmute

RaidMax commented on Plugins/Mute/MuteManager.cs:
```diff
@@ -98,11 +103,13 @@ public async Task<bool> Unmute(Server server, EFClient origin, EFClient target,
         if (!target.IsIngame && clientMuteMeta.MuteState is MuteState.Unmuting) return false;
 
-        if (clientMuteMeta.MuteState is not MuteState.Unmuting)
+        if (clientMuteMeta.MuteState is not MuteState.Unmuting || origin.ClientId != 1)
```
Should this be an ||, or an &&?

RaidMax commented on Plugins/Mute/MuteManager.cs:
```diff
@@ -130,14 +137,30 @@ public async Task<bool> Unmute(Server server, EFClient origin, EFClient target,
+        await using var context = _databaseContextFactory.CreateContext();
+        var mutePenalties = await context.Penalties
+            .Where(penalty => penalty.OffenderId == client.ClientId &&
+                penalty.Type == EFPenalty.PenaltyType.Mute || penalty.Type == EFPenalty.PenaltyType.TempMute &&
+                penalty.Expires == null || penalty.Expires > DateTime.UtcNow)
```
I believe you need to apply parenthesis for correct order of operations
```csharp
(penalty.Type == EFPenalty.PenaltyType.Mute || penalty.Type == EFPenalty.PenaltyType.TempMute) &&
                (penalty.Expires == null || penalty.Expires > DateTime.UtcNow)
```

RaidMax commented on WebfrontCore/Controllers/Client/ClientController.cs:
```diff
@@ -162,7 +162,12 @@ public async Task<IActionResult> Profile(int id, MetaType? metaFilterType, Cance
             }
 
-            clientDto.ActivePenalty = activePenalties.OrderByDescending(_penalty => _penalty.Type).FirstOrDefault();
+            clientDto.ActivePenalty = activePenalties.MaxBy(penalty => penalty.Type switch
```
reviewing this again we should add a comment why we're doing this.

Ayymoss committed:
Resolves issues in code review
Added comment in ClientController.cs
Fixed order of operations in MuteManager.cs
Fixed condition in MuteManager.cs

Ayymoss committed:
Resolves issues in code review
Added comment in ClientController.cs
Fixed order of operations in MuteManager.cs
Fixed condition in MuteManager.cs

Ayymoss committed:
Merge branch 'release/pre' of https://github.com/MrAmos123/IW4M-Admin into release/pre

Ayymoss committed:
Changed localisation value to be more generic
Fix null reference warning (it should never be null) (34da216)
